src/shared/parameter_v2/config/examples/example.yaml

```diff
@@ -0,0 +1,140 @@
+# This file provides an example for how to define a config file, including examples
+# of all parameter types, special attributes, and how to include other configs
+#
+# Parameter schema
+#
+# type:
+#   name:
+#   value:
+#   description:
+#   other attributes...
+#
+# Notes:
+# - All 'name' attributes must be unique per-file
+# - All 'name' attribute values should be snake_case. Spaces are not allowed.
+# - All config file names must be unique
+
+#########################################################################################
+# The following are examples of how to include other configs
+#########################################################################################
+# Config files can include other configs. This is how the parameter hierarchy is built
+# up, and allows us to share parameters between configs. For example, this makes it
+# easy to define some passing parameters in one location, and share those parameters
+# with the subset of Plays that need them.
+#
+# All included config files need to be listed under a single `include` section.
+# Because all config filenames must be unique, just the filename needs to be
+# specified (not the path).
+#
+# In the generated code, included configs and their attributes can be accessed by
+# the name of the included config.
+# Eg. foo.foo_bool
+include:
+  - "foo.yaml"
+---
+
+#########################################################################################
+# The following are example of every type of parameter that can be used
+#########################################################################################
+- bool:
+    name: example_bool_param
+    value: true
+    description: >
+      Can be true or false
+- int:
+    name: example_bool_param
+    min: 0
+    max: 5
+    value: 3
+    description: >
+      Can be any integer value in the range [min, max]
+- uint:
+    name: example_unsigned_int_param
+    min: 0 # Must be >= 0
+    max: 5
+    value: 3
+    description: >
+      Can be any integer value in the range [min, max].
+      The minimum value must be >= 0
+- float:
+    name: example_float_param
+    min: 1.1
+    max: 9.01
+    value: 4.04
+    description: >
+      Can be any value in the range [min, max]
+- double:
+    name: example_double_param
+    min: 0
+    max: 5.4
+    value: 0.2
+    description: >
+      Can be any value in the range [min, max]
+
+      Will not be generated for C code, because we assume that is firmware code
+      that cannot represent doubles.
```

Comment:
Why not just have a single floating point type that's represented as a `float` on the `C` side and a double on the `C++` side?